stdlib/socket/src/main/ballerina/socket/client_socket.bal

```diff
@@ -55,6 +55,15 @@ public type Client client object {
     # + return - - number of bytes got written or an error if encounters an error while writing
     public remote extern function write(byte[] content) returns int|error;
 
+    # Reads data from the client socket. If the length specified, then wait until received that many bytes from the client or
+    # else return what ever data available in OS buffer. In the case of connection close by client then return either -1
+    # or currently available data in the buffer.
+    # Number of bytes returned will be < 0 if the client closes the connection.
+    #
+    # + length - - Positive integer. Represents the number of bytes which should be read
+    # + return - - Content as byte array and the number of bytes read or an error if encounters an error while reading
+    public remote extern function read(int length = -100) returns (byte[], int)|error;
+
     # Closes the client socket connection.
     #
     # + return - - an error if encounters an error while closing the connection or returns nil otherwise
```

Comment:
returns the error if an error is encountered while closing the connection. Otherwise, returns nil.